perf/src/sigverify.rs

```diff
@@ -622,33 +629,47 @@ pub fn ed25519_verify_cpu(batches: &mut [PacketBatch], reject_non_vote: bool, pa
                 }
             })
         });
-    } else if desired_thread_count < get_thread_count() {
-        // Dynamically compute minimum packet length to spread the load while minimizing threads.
-        let packets_per_thread = packet_count.saturating_div(desired_thread_count);
-        PAR_THREAD_POOL.install(|| {
-            batches
-                .into_par_iter()
-                .flatten()
-                .collect::<Vec<&mut Packet>>()
-                .into_par_iter()
-                .with_min_len(packets_per_thread)
-                .for_each(|packet: &mut Packet| {
-                    if !packet.meta.discard() && !verify_packet(packet, reject_non_vote) {
-                        packet.meta.set_discard(true);
-                    }
-                })
-        });
+        1
     } else {
-        // When using all available threads, skip the overhead of flattening, collecting, etc.
-        PAR_THREAD_POOL.install(|| {
-            batches.into_par_iter().for_each(|batch: &mut PacketBatch| {
-                batch.par_iter_mut().for_each(|packet: &mut Packet| {
-                    if !packet.meta.discard() && !verify_packet(packet, reject_non_vote) {
-                        packet.meta.set_discard(true);
-                    }
-                })
+        let packet_verification_closure = |packet: &mut Packet| {
+            if !packet.meta.discard() && !verify_packet(packet, reject_non_vote) {
+                packet.meta.set_discard(true);
+            }
+            if let Some(thread_mask) = CheckedShl::checked_shl(
+                &1,
+                rayon::current_thread_index().unwrap().try_into().unwrap(),
+            ) {
+                thread_in_use_multihot.fetch_or(thread_mask, Ordering::Relaxed);
+            }
```

Comment:
wouldn't this add extra overhead to packet verification?
Also wouldn't this break if there are more than 64 threads?

Reply:
> I don't believe we have any cases where we are running with that many threads, 

You can check `cpu-stats` metrics

Reply:
Highest core count is 96 on testnet, which I believe would mean max of 48 threads for Sigverify. Either way, future proofing is good